shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Bbody.hs

```diff
@@ -80,10 +80,11 @@ bbodyTransition
      , Signable (DSIGN crypto) (TxBody crypto)
      )
   => TransitionRule (BBODY crypto)
-bbodyTransition = do
-  TRC ( BbodyEnv oslots pp _reserves
-      , BbodyState ls b
-      , Block (BHeader bhb _) txsSeq@(TxSeq txs)) <- judgmentContext
+bbodyTransition = judgmentContext >>=
+  \(TRC ( BbodyEnv oslots pp _reserves
+          , BbodyState ls b
+          , Block (BHeader bhb _) txsSeq@(TxSeq txs))
+   ) -> do
```

Comment:
Particular reason for this change in this PR? Not that it's a problem...

Reply:
It's to work around a [bug](https://gitlab.haskell.org/ghc/ghc/issues/15681).

Matching on a pattern synonym inside a do block always desugars to code with a `MonadFail` constraint, even in the presence of a COMPLETE pragma.